### DISCUSSION TITLE
Potential bad practice with secret key in tutorial example

### DISCUSSION THREAD

**Discussion by FavoritoHJS** [pallets/flask#5501] [Q&A]
So, while looking at the tutorial, I noticed something odd: The page describing the secret key (https://flask.palletsprojects.com/en/3.0.x/config/#SECRET_KEY) says to not commit the secret key, but the tutorial puts the secret key in __init__.py, which is almost certainly gonna be commited.

**Comment by FavoritoHJS** [comment]
and as per tradition, i think of a solution mere seconds after publishing my question
it doesn't really matter if it's commited, so long as the repo is fully private. Once you share it is where problems arise.

**Reply by ThiefMaster** [reply]
No, you should never commit secrets to a repo (especially one that's not used ONLY for secrets, config data, etc.), not even to a private one.

**Comment by ThiefMaster** [comment]
I think the point of the tutorial is to have something simple. Adding a config wile would make it more complex...
